src/libhirte/common/cfg.c

```diff
@@ -132,16 +132,29 @@ int cfg_load_complete_configuration(
                                 default_config_file,
                                 strerror(-result));
                         return result;
+                } else if (result > 0) {
+                        fprintf(stderr,
+                                "Error parsing configuration file '%s' on line %d\n",
+                                default_config_file,
+                                result);
+                        return -EINVAL;
                 }
         }
 
         if (custom_config_file != NULL) {
                 result = cfg_load_from_file(config, custom_config_file);
-                if (result < 0 && result != -ENOENT) {
```

Comment:
Haven't had the case yet, but since we didn't handle `r > 0` (indicating parsing at line r failed) means we "skipped" an invalid configuration file.